packages/main/src/Switch.ts

```diff
@@ -212,22 +212,25 @@ class Switch extends UI5Element {
 		}
 
 		if (isEnter(e)) {
-			this.toggle();
+			this._onclick();
 		}
 	}
 
 	_onkeyup(e: KeyboardEvent) {
 		if (isSpace(e)) {
-			this.toggle();
+			this._onclick();
 		}
 	}
 
 	toggle() {
 		if (!this.disabled) {
-			this.checked = !this.checked;
-			this.fireEvent("change");
+			const changePrevented = !this.fireEvent("change", null, true);
```

Comment:
when the event is fired, the value should already be updated like before so the app can decide whether the default should be prevented.

in case the event is prevented, the old value is restored in the `if` statement below